### PR TITLE
Allow custom format for Date, DateTime and Time

### PR DESCRIPTION
**Pull request by LIQRGV**
Also change some test

### CODE REVIEW

**Comment by nicolasblanco** [comment]
Hi there!

Seems cool, may you merge master?

**Comment by LIQRGV** [comment]
Sadly, it doesn't pass Travis CI.
That's strange.
:(

**Comment by nicolasblanco** [comment]
I've fixed the build issue on master, you just need to merge master branch again 👍

**Comment by LIQRGV** [comment]
Well, the code pass the test already.
Cool.
Thank you.
:D

**Comment by numbcoder** [comment]
merged?

**Comment by LIQRGV** [comment]
I cant merge to master, 
so, sorry.
Ask the owner instead.

**Comment by LIQRGV** [comment]
@nicolasblanco could you merge this commit to master ?

File: lib/rails_param/param.rb [hunk 1]
```diff
@@ -83,9 +83,13 @@ def coerce(param, type, options = {})
         return Integer(param) if type == Integer
         return Float(param) if type == Float
         return String(param) if type == String
-        return Date.parse(param) if type == Date
-        return Time.parse(param) if type == Time
-        return DateTime.parse(param) if type == DateTime
+        if [Date, DateTime, Time].include? type
```
Comment (iMacTia) [inline lib/rails_param/param.rb]: This is generating the `[Date, DateTime, Time]` array every time we check a parameter.
Please move this into a frozen constant and use that one instead.

File: lib/rails_param/param.rb [hunk 2]
```diff
@@ -114,8 +118,8 @@ def validate!(param, options)
                                                                                       param.nil?
                                                                                   end
           when :format
-            raise InvalidParameterError, "Parameter must be a string if using the format validation" unless param.kind_of?(String)
-            raise InvalidParameterError, "Parameter must match format #{value}" unless param =~ value
+            raise InvalidParameterError, "Parameter must be a string if using the format validation" unless [String, Date, DateTime, Time].any? { |cls| param.kind_of? cls }
```
Comment (iMacTia) [inline lib/rails_param/param.rb]: Same as above, this is quite inefficient and could be optimised.

**Comment by LIQRGV** [comment]
I will check.
Sorry for not test before push.

**Comment by LIQRGV** [comment]
Done.
It pass travis already.

**Comment by iMacTia** [comment]
Amazing, thanks @LIQRGV.
Added to release v0.11.0 👍

**Pull request merged**